### PR TITLE
ag scrub the extra field

### PR DESCRIPTION
**Pull request by AGarrow**
The extra field is not being scrubbed according to the scrub fields, this could lead to PII leaking internally as many would assume that the extra params are scrubbed.

in particular, calling `Rollbar.error(e, extra_params: some_request_params)` for example.

### CODE REVIEW

**Comment by jessewgibbs** [comment]
@AGarrow thanks for the PR.

@waltjones can you take a look?

File: lib/rollbar/item.rb [hunk 1]
```diff
@@ -157,12 +157,23 @@ def build_backtrace_body
 
     def build_extra
       if custom_data_method? && !Rollbar::Util.method_in_stack(:custom_data, __FILE__)
-        Util.deep_merge(custom_data, extra || {})
+        Util.deep_merge(custom_data, scrub_extra || {})
       else
         extra
```
[the remark this answers is not in the document]
Comment (AGarrow) [reply lib/rollbar/item.rb]: @waltjones no problem, thanks for your response!

on line 162 I ran into issues with circular dependencies as it seemed to break rollbar errors.

 I don't have a reason not to scrub `custom_data` other than that it wasn't what was leaking PII for us, but I can add that too.

[the remark this answers is not in the document]
Comment (AGarrow) [reply lib/rollbar/item.rb]: is there an easy way for me to detect if what's being handled is an internal rollbar error and just return `extra` in that case?

**Comment by AGarrow** [comment]
@waltjones turns out the params scrubber was susceptible to cyclical references, I think I've handled that now. I chose to prioritize runtime over memory in that case, I hope thats appropriate.

File: lib/rollbar/scrubbers/params.rb [hunk 2]
```diff
@@ -56,6 +57,9 @@ def build_whitelist_regex(whitelist)
       end
 
       def scrub(params, options)
+        return params if @scrubbed_object_ids[params.object_id].present?
```
Comment (waltjones) [inline lib/rollbar/scrubbers/params.rb]: `#present?` isn't guaranteed to be available in non-Rails host apps. (Rack, Sinatra) 
```
return params if @scrubbed_object_ids[params.object_id]
```
is best here.

File: lib/rollbar/scrubbers/params.rb [hunk 3]
```diff
@@ -20,6 +20,7 @@ def self.call(*args)
       def call(options = {})
         params = options[:params]
         return {} unless params
```
Comment (waltjones) [inline lib/rollbar/scrubbers/params.rb]: Rubocop wants an empty line after this guard clause.

> Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.

**Comment by waltjones** [comment]
@AGarrow thank you!

**Comment by AGarrow** [comment]
@waltjones no problem! do you have a rough idea of when the next release might be?

**Comment by waltjones** [comment]
I think @ArturMoczulski is working on a release any day now.

**Comment by AGarrow** [comment]
👍 thank you!

**Pull request merged**